JumpscaleLibs/sal/zosv2/billing.py

```diff
@@ -1,59 +1,33 @@
-        :return: list of ResourceUnitsNode object
-        :rtype: list
-        """
-        rp = ResourceParser(self._explorer, reservation)
-        return rp.calculate_used_resources()
-
-    def reservation_resources_cost(self, reservation):
-        """
-        compute how much resource units is reserved in the reservation
-
-        :param reservation: reservation object
-        :type reservation: tfgrid.workloads.reservation.1
-        :return: list of ResourceUnitsNode object with costs filled in
-        :rtype: list
-        """
-        rp = ResourceParser(self._explorer, reservation)
-        return rp.calculate_used_resources_cost()
 
-    def payout_farmers(self, client, reservation):
+class Billing:
+    def payout_farmers(self, client, reservation_response):
         """
         payout farmer based on the resources per node used
-
-        :param client: tfchain wallet or stellar client
-        :type client: a wallet of j.clients.tfchain or a client of j.client.stellar
-        :param reservation: reservation object
-        :type reservation: tfgrid.workloads.reservation.1
-        :return: list of transactions
-        :rtype: list
-        """
-        rp = ResourceParser(self._explorer, reservation)
-        costs = rp.calculate_used_resources_cost()
-        return rp.payout_farmers(client, costs, reservation.id)
-
-    def verify_payments(self, client, reservation):
-        """
-        verify that a reservation with a given ID has been paid for, for all farms belonging to the current user 3bot
-
-        :param client: tfchain wallet or stellar client
-        :type client: a wallet of j.clients.tfchain or a client of j.client.stellar
-        :param reservation: reservation object
-        :type reservation: tfgrid.workloads.reservation.1
-        :return: if the reservation has been fully funded for the farms owned by the current user 3bot
-        :rtype: bool
-        """
-        rp = ResourceParser(self._explorer, reservation)
-        return rp.validate_reservation_payment(client, reservation.id)
+        :param client: stellar client
+        :type client: j.clients.stellar
+        :param reservation_response: reservation register response
+        :type reservation_response: reservation register response
+        """
+        asset = None
+        if client.network == "TEST":
+            asset = ASSET_CODE + ":" + TFT_ISSUER_TEST
+        else:
+            asset = ASSET_CODE + ":" + TFT_ISSUER_PROD
+
+        reservation_id = reservation_response["id"]
+        escrow_informations = reservation_response["escrow_information"]
+        for escrow in escrow_informations:
+            escrow_address = escrow["escrow_address"]
+            total_amount = escrow["total_amount"] / 10e6
+            try:
+                client.transfer(escrow_address, total_amount, asset=asset, memo_text=str(reservation_id))
+            except BadRequestError as e:
+                self._log_debug(e)
```

Comment:
What does this exception means? I do not like we mask errors like this, better to just let it bubble up